### DISCUSSION TITLE
osgEarthTriton problem : undefined function "float oe_terrain_getElevation();"

### DISCUSSION THREAD

**Discussion by pskpg** [pelicanmapping/osgearth#2316] [Support Q&A] [answered]
Hi, Glenn!
When I run osgearth_triton.exe with a local bathemetry, cmd shows this:

![image](https://github.com/gwaldron/osgearth/assets/20101178/2f6d02ae-b038-4f9b-b804-823fa1d8e134)

And the ocean doesn't show as excepted.

The oe_terrain_getElevation is declared in TritonHeightMap.cpp with no implementation, as follows:

![image](https://github.com/gwaldron/osgearth/assets/20101178/19419f70-98ce-4d5d-9b5a-d53f42d1f812)

How to solve this problem?
Thanks!

**Comment by gwaldron** [comment]
Is this causing things to not work, or is it just an annoying message?
I ask because sometimes the shader composition framework will display quirks like this but the code will still work.
The function oe_terrain_getElevation() exists in the SDK shader which should be linked with the shader you are seeing.

**Comment by fkane** [comment, marked as answer]
Perhaps you missed the step of replacing Triton's "user shaders" with the ones provided with osgEarth? See https://github.com/gwaldron/osgearth/tree/master/src/osgEarthTriton/Shaders and https://www.sundog-soft.com/docs/triton/gettingstarted.html#osgearth

Also be aware you need to build osgEarth from source for Triton integration to work properly; the current osgEarth 3.4 release does not have some important fixes you need.
